public/scripts.js

```diff
@@ -8,36 +8,164 @@
     })
   }
 
-  const getVouchers = function () {
-    return $.get('/vouchers.json')
+  const ProductModel = function (price, countChangeHandler) {
+    this.price = 0
+    this.count = 1
+
+    const $productsCountInput = $('#products-count')
+
+    this.init = function (price) {
+      const _self = this
+      _self.price = price
+      _self.count = $productsCountInput.val()
+      _self.countChangeHandler = countChangeHandler
+
+      $productsCountInput.on('change', function () {
+        _self.count = $(this).val()
+        _self.countChangeHandler(_self.count, _self.price)
+      })
+
+      return _self
+    }
+
+    this.getPrice = function () {
+      return this.price
+    }
+
+    this.getCount = function () {
+      return this.count
+    }
+
+    return this.init(price)
   }
 
-  Voucherify.initialize(clientConfig.clientApplicationId, clientConfig.clientPublicKey)
-  Voucherify.setIdentity('gustav@purpleson.com')
-  Voucherify.render('#voucher-checkout', {
-    textPlaceholder: 'e.g. Testing7fjWdr',
-    onValidated: function (res) {
-      const voucherCode = $('#voucher-checkout .voucherifyCode').val()
-      const trackingId = res.tracking_id
-
-      if (res.valid && confirm('Czy chcesz wykorzystać ten Voucher?')) {
```

Comment:
1. onValidated is called only if result is valid (see https://github.com/rspective/voucherify.js/blob/master/lib/voucherify.js#L346)
2. confirm message in Polish?